src/ert/gui/tools/plot/plot_window.py

```diff
@@ -226,19 +213,27 @@ def currentPlotChanged(self):
 
             plot_context.log_scale = key_def.log_scale
 
-            plot_widget.updatePlot(plot_context, ensemble_to_data_map, observations)
+            if self._preferred_ensemble_x_axis_format == PlotContext.UNKNOWN_AXIS:
+                for _, data in ensemble_to_data_map.items():
+                    data = data.T
```

Comment:
Why is this transpose needed? For the check below? Not seeing it used elsewhere

Reply:
This was needed for the check below yes.
I've slightly refactored this to be somewhat more optimised, since we will encounter date-time objects for the most part.